server/controllers/request.js

```diff
@@ -0,0 +1,83 @@
+const Request = require("../models/Request");
+const asyncHandler = require("express-async-handler");
+
+// @route GET /requests
+// @desc get requests for logged in user
+// @access Private
+exports.getRequests = asyncHandler(async (req, res) => {
+  const requests = await Request.find({
+    sitterId: req.user.id,
+  });
+
+  res.status(200).json({
+    success: { requests },
+  });
+});
+
+// @route POST /requests
+// @desc Create a new request
+// @access Private
+exports.createRequest = asyncHandler(async (req, res) => {
+  const { sitterId, start, end } = req.body;
+
+  const propertyIsMissing = !sitterId || !start || !end;
+
+  if (propertyIsMissing || !mongoose.Types.ObjectId.isValid(sitterId)) {
```

Comment:
Note (no action needed): In the future we can move this into a middleware (using express validator - already installed) and apply it to the relevant route - it would make the code a lot cleaner and more maintainable